lib/src/utils/serialization_utils.dart

```diff
@@ -128,6 +137,20 @@ class SerializationUtils {
     });
   }
 
+  static Uint8List serializeCoin(String coin) {
+    return useBinaryWriter((ByteDataWriter writer) {
+      writeSerializedUint64(writer, BigInt.parse(coin));
+      return writer.toBytes();
+    });
+  }
```

Comment:
Cateva chestii trebuie schimbate aici:
a) `coin` ar trebui sa fie `BigInt` nu `String` din start
b) `Uint64` nu e destul. Valoare pentru coin nu are limita. Deci trebuie serializat fara sa limitam valoarea la uint64.

Reply:
Esti sigur ca acest coin nu e serializat direct ca String in ts?

De ex: Daca serializezi numarul 20 ca Uint sau "20" ca string o sa dea un rezultat complet diferit.

Reply:
in ts e `export type Uint64_str = string & {__type: 'uint64_t'}`  , el foloseste functia `export function serializeCoin(coin: Uint64_str): Buffer {
  return Buffer.concat([uint64_to_buf(coin)])
}` care converteste un string ce reprezinta un numar intreg pe 64 de biti intr-un Buffer

Reply:
Hmm, ok. Atunci lasam implementarea in mare parte la fel. Dar `coin` tot trebuie schimbat in `BigInt` din `String`